# Generated by Django 4.0.3 on 2022-03-28 23:19

from django.conf import settings
import django.core.validators
from django.db import migrations, models
import django.db.models.deletion
import users.validators


class Migration(migrations.Migration):

    initial = True

    dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]

    operations = [
        migrations.CreateModel(
            name='Profile',
            fields=[
                ('id', models.CharField(max_length=32, primary_key=True, serialize=False, unique=True)),
                ('username', models.CharField(error_messages={'unique': 'A user with that username already exists.'}, help_text='Required. 32 characters or fewer. Lowercase letters, digits _ only; must start with a letter.', max_length=32, unique=True, validators=[django.core.validators.MinLengthValidator(4), users.validators.MyUsernameValidator()])),
                ('name', models.CharField(max_length=64, null=True)),
                ('lastname', models.CharField(max_length=64, null=True)),
                ('type', models.CharField(choices=[('STD', 'Standard'), ('ADM', 'Admin')], default='STD', max_length=3)),
                ('is_verified', models.BooleanField(default=False)),
                ('bio', models.TextField(max_length=512, null=True)),
                ('sign_in_provider', models.CharField(max_length=32)),
                ('created_at', models.DateTimeField(auto_now_add=True)),
                ('updated_at', models.DateTimeField(auto_now=True)),
                ('user', models.OneToOneField(on_delete=django.db.models.deletion.PROTECT, to=settings.AUTH_USER_MODEL)),
            ],
        ),
    ]
